packages/muelu/test/unit_tests/InverseApproximationFactory.cpp

```diff
@@ -0,0 +1,215 @@
+      invapproxFact->SetFactory("A",MueLu::NoFactory::getRCP());
+      invapproxFact->SetParameter("inverse: approximation type", Teuchos::ParameterEntry(std::string("diagonal")));
+
+      // request InverseApproximation operator
+      level.Request("Ainv", invapproxFact.get());
+
+      // generate Schur complement operator
+      invapproxFact->Build(level);
+
+      RCP<Vector> Ainv = level.Get<RCP<Vector> >("Ainv", invapproxFact.get());
+      TEST_EQUALITY(Ainv.is_null(), false);
+      TEST_EQUALITY(Ainv->getMap()->getMinGlobalIndex(), comm->getRank() * int(n*n/comm->getSize()));
+      TEST_EQUALITY(Ainv->getMap()->getMaxGlobalIndex(), comm->getRank() * int(n*n/comm->getSize()) + int(n*n/comm->getSize()-1));
+
+      Teuchos::ArrayRCP<const Scalar> AinvData = Ainv->getData(0);
+      bool bCheck = true;
+      for(int i=0; i<int(n*n/comm->getSize()); i++) if(AinvData[i] != Teuchos::as<Scalar>(0.25)) bCheck = false;
+      TEST_EQUALITY(bCheck, true);
+    }
+
+  } //InverseDiagonalConstructor
+
+  TEUCHOS_UNIT_TEST_TEMPLATE_4_DECL(InverseApproximationFactory, InverseLumpingConstructor, Scalar, LocalOrdinal, GlobalOrdinal, Node)
+  {
+#   include "MueLu_UseShortNames.hpp"
+    MUELU_TESTING_SET_OSTREAM;
+    MUELU_TESTING_LIMIT_SCOPE(Scalar,GlobalOrdinal,NO);
+    out << "version: " << MueLu::Version() << std::endl;
+
+    RCP<const Teuchos::Comm<int> > comm = Parameters::getDefaultComm();
+    Xpetra::UnderlyingLib lib = MueLuTests::TestHelpers::Parameters::getLib();
+
+    {
+      const int n = 20;
+      Teuchos::RCP<Matrix> A = MueLuTests::TestHelpers::TestFactory<Scalar,LocalOrdinal,GlobalOrdinal,Node>::Build1DPoisson(n, lib);
+      A->scale(0.5);
+
+      Level level;
+      TestHelpers::TestFactory<Scalar,LocalOrdinal,GlobalOrdinal,Node>::createSingleLevelHierarchy(level);
+      level.Set("A", A);
+
+      RCP<InverseApproximationFactory> invapproxFact = rcp( new InverseApproximationFactory() );
+      invapproxFact->SetFactory("A",MueLu::NoFactory::getRCP());
+      invapproxFact->SetParameter("inverse: approximation type", Teuchos::ParameterEntry(std::string("lumping")));
+
+      // request InverseApproximation operator
+      level.Request("Ainv", invapproxFact.get());
+
+      // generate Schur complement operator
+      invapproxFact->Build(level);
+
+      RCP<Vector> Ainv = level.Get<RCP<Vector> >("Ainv", invapproxFact.get());
+      TEST_EQUALITY(Ainv.is_null(), false);
+      TEST_EQUALITY(Ainv->getMap()->getMinGlobalIndex(), comm->getRank() * int(n/comm->getSize()));
+      TEST_EQUALITY(Ainv->getMap()->getMaxGlobalIndex(), comm->getRank() * int(n/comm->getSize()) + int(n/comm->getSize()-1));
+
+      Teuchos::ArrayRCP<const double> AinvData = Ainv->getData(0);
+      bool bCheck = false;
+      for(int i=0; i<int(n/comm->getSize()); i++)
+        if(std::abs(AinvData[i] - 0.66666666667) < 1e-8 || std::abs(AinvData[i] - 0.5) < 1e-8)
```

Comment:
Does this pass for float and complex<double>?

Reply:
That's actually a good point. For `float` I would say yes, but for `complex` I doubt it. I'll have a look into it.

Reply:
Could you try it with `float`? It's not enabled by the autotester..

Reply:
On my local build it passes with `float` set via `-D Trilinos_ENABLE_FLOAT:BOOL=ON`.